### ISSUE TITLE
indent expects wrong indentation for logical/arithmetic expressions

### ISSUE THREAD

**Issue by facboy** [eslint/eslint#6731]
**What version of ESLint are you using?**

3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
indent: [
    'error',
    4,
    {
        SwitchCase: 1
    }
]
```

**What did you do? Please include the actual source code causing the issue.**

```js
(function() {
    var x = 0 ||
        {
            a: 1,
            b: 2
        };

    var y = x ||
        true;
})();
```

**What did you expect to happen?**

I expected it not to have an error about the indentation on lines 3-6

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
/home/user/test.js
  3:13  error  Expected indentation of 4 space characters but found 8   indent
  4:17  error  Expected indentation of 8 space characters but found 12  indent
  5:17  error  Expected indentation of 8 space characters but found 12  indent
  6:13  error  Expected indentation of 4 space characters but found 8   indent
  8:9   error  'y' is defined but never used                            no-unused-vars

✖ 5 problems (5 errors, 0 warnings)
```

**Comment by vitorbal** [comment]
Thanks for reaching out, @facboy!

I can reproduce this in the demo. Just to be clear for other readers, the second `var` declaration in the example does **not** trigger any error messages, so it might have something to do with the way the `indent` rule deals with ternary conditions and object literals.

@ilyavolodin may be of interest for you because of the `indent` rule improvement coordination.

**Comment by alecharmon** [comment]
Hey all I'm new to the community but I'll try my hand at taking this on if there are objections.

**Comment by ilyavolodin** [comment]
@alecharmon We would definitely appreciate a pull request for this issue. However, keep in mind, this is most likely part of the multiline issue with indent rule, which is much larger. You can search open issues for other reports related to multiline indentation problems.

**Comment by sstern6** [comment]
@alecharmon @ilyavolodin There hasnt been any movement on this one in about a month.  Would like to open a PR unless @alecharmon has already started working on it,  but brand new to the code base.  If its ok to open a PR could you please point me in the right direction for relevant files for this issue and will research issues related to multiline issues. 

Thank you

**Comment by alecharmon** [comment]
Hey @sstern6 @ilyavolodin im still picking this up, life has been weird recently but I'm back on the horse! will get some code up in a pr soon!

**Comment by alecharmon** [comment]
code is ready for a pr, will work on getting that up tomorrow
